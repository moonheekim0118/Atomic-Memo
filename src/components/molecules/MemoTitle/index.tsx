import React from 'react';
import styled from 'styled-components';
import Span from '../../atoms/Span';
import MemoKind from '../../../util/memoKind';

interface Props {
  title: string;
  kind: string;
}

const MemoTitle = ({ title, kind }: Props) => {
  return (
    <StyledMemoTitle>
      <Span size="small" padding="0 0.5em 0 0">
        {MemoKind[kind]}
      </Span>
      <Span size="small" width="5.5em">
        {title}
      </Span>
    </StyledMemoTitle>
  );
};

const StyledMemoTitle = styled.div`
  display: flex;
  padding: 2em;
`;

export default MemoTitle;
