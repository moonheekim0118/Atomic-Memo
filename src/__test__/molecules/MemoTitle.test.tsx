import 'jsdom-global/register';
import React from 'react';
import MemoTitle from '../../components/molecules/MemoTitle';
import { mount } from 'enzyme';

describe('<MemoTitle/>', () => {
  let container = null;

  it('renders correctly', () => {
    container = mount(<MemoTitle title="test" kind="Work" />);
  });

  it('matches snpashot', () => {
    expect(container.html()).toMatchSnapshot();
  });
});
