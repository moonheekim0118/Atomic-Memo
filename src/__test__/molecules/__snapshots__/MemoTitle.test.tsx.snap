// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MemoTitle/> matches snpashot 1`] = `"<div class=\\"MemoTitle__StyledMemoTitle-e33wpr-0 cPfMRA\\"><span width=\\"auto\\" display=\\"inline\\" class=\\"Span__StyledSpan-sc-10y5m6u-0 dZpPqp small\\"></span><span width=\\"5.5em\\" display=\\"inline\\" class=\\"Span__StyledSpan-sc-10y5m6u-0 bvWVQT small\\">test</span></div>"`;
